script.js

```diff
@@ -1,4 +1,159 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+/*
+There are 2 players and players take turns.
+When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+How many game modes:
+1. Player 1 click submit to roll dice.
+2. Player 1 click 1 or 2 to choose order of dice. 
+3. Player 2 click submit to roll dice. 
+4. Player 2 click 1 or 2 to choose order of dice. 
+5. Determine who gets the higher combined number.  
+*/
+
+//global variables for current game mode
+var playerRoll1 = 0;
+var playerRoll2 = 0;
+var gameMode = "start game by rolling 2 dice";
+var currentPlayer = 1;
+var allDiceOrder = [];
+var accumulativeRound = 0;
```

Comment:
nice and clear global variables. i would say best to have a comment on what `allDiceOrder` and `accumulativeRound` tracks, as it can be a bit unclear. alternatively, rename the variables to be something more descriptive then  the comment is unnecessary. this is another point for why descriptive variable names are desired as your code becomes self-documenting

Reply:
another point here: i noticed you have 4 game modes: "start game by rolling 2 dice", "player choose dice order", "next player roll 2 dice", and "determine winner among 2 players"

You typed out the string values of these game modes explicitly in your `beatThatGame` and `main` function, a better practice would be to define them as global variables and in your functions, reference them using the variables when you need to check for what the game mode is. To illustrate:

```
const START_GAME = "start game by rolling 2 dice";
const NEXT_PLAYER = ""next player roll 2 dice";
const CHOOSE_ORDER = "player choose dice order";
const DETERMINE_WINNER = "determine winner among 2 players";
var gameMode = START_GAME;
```
This goes together with your other global variables here, then for example in line 133 when you check `if (gameMode == "next player roll 2 dice")` you can do `if (gameMode == NEXT_PLAYER)` instead. Likewise, when you are updating the game mode in example line 122, `gameMode = "player choose dice order"` you do `gameMode = CHOOSE_ORDER;` instead.

This reduces code repetition as you only need to use the string value once which prevents errors arising from typos. I am also using the `const` keyword to declare them rather than var. For constants, some people will use SCREAM_CASE as well for the variable name. I know this wasn't really covered in the course but it's a minor step up if you're interested.

For more info:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const